dm/pkg/cputil/table.go

```diff
@@ -23,6 +31,16 @@ func LightningCheckpoint(task string) string {
 	return task + "_lightning_checkpoint_list"
 }
 
+// LightningCheckpointSchema returns lightning's checkpoint schema name.
+func LightningCheckpointSchema(task string, sourceID string) string {
+	if task == "" {
+		return "tidb_lightning_checkpoint"
+	}
+	return utils.TruncateStringQuiet(
+		fmt.Sprintf("%s_%d_tidb_lightning_checkpoint", task, utils.GenHashKey(sourceID)),
```

Comment:
it's OK to left sourceID in database name? why task is not hashed but sourceID is hashed

Reply:
-- not sure if lightning escape the database name.
![image](https://user-images.githubusercontent.com/17703670/224595371-d1bb05ff-06b7-4c54-8d03-af378992b445.png)

-- it's OK to left sourceID in database name? why task is not hashed but sourceID is hashed
For cloud DM, 'task' alway be the number like '12123', so the previous table name use the 'task' directly. 
'sourceID' may be the user-defined identifier which can contains some illegal chars for 'schema' and 'table'.

Reply:
After test, current implement doesn't escape the schema name.

Reply:
Currently, 'sourceID' is like  '{\"SourceID\":\"source-990001\"'. May be we can directly escape the name.

Reply:
Will task contain special character? we might also need to hash it

Reply:
there's no task param in engine `create-job` api, it's generated by engine